File: linkedlist/LinkListLength/LinkListLength.py
# count number of elements in a link List

class Node: 
    def __init__ (self,data):
        self.data = data
        self.next = None #init next pointer to next object as null

class LinkListLength:
    # Function to init head 
    def __init__(self):
        self.head = None 

    def push(self,new_data):
        new_node = Node(new_data)
        new_node.next = self.head
        self.head = new_node

    def getCount(self): 
        temp = self.head #init temp 
        count = 0 
        while (temp):
            count += 1
            temp = temp.next
        return count

if __name__ == '__main__':
    linklist = LinkListLength()
    linklist.push(1)
    linklist.push(3)
    linklist.push(1)
    linklist.push(2)
    linklist.push(1)
    print ("Count of nodes is :",linklist.getCount()) 

File: linkedlist/LinkListLength/LinkListLength.java
// Count number of elements in a link list 


class Node { 
	int data; 
	Node next; 
	Node (int d){
		data = d ; 
		next = null; 
	}
}

public class LinkListLength {

	Node head;  // dead of list 
	
	public void push(int new_data ) {
		Node new_node = new Node(new_data);
		new_node.next = head; 
		head = new_node; 
	}
	
	public int getCount() { 
		Node temp = head; 
		int count = 0; 
		while (temp != null ) {
			count++; 
			temp = temp.next; 
		}
		return count; 
	}
	
	public static void main(String [] args) {
		LinkListLength linklist = new LinkListLength();
		linklist.push(1);
		linklist.push(3);
		linklist.push(1);
		linklist.push(2);
		linklist.push(1);
		System.out.println("count of nodes is: " + linklist.getCount());
		
		
	}
	
	
	
}
